agent/consul/internal_endpoint.go

```diff
@@ -323,18 +323,20 @@ func (m *Internal) GatewayServices(args *structs.ServiceSpecificRequest, reply *
 			var index uint64
 			var services structs.GatewayServices
 
-			// We only use this call to validate the RPC call, don't add the watch set
-			_, ingressCfg, err := state.ConfigEntry(nil, structs.IngressGateway, args.ServiceName, &args.EnterpriseMeta)
-			if err != nil {
-				return err
-			}
-			// We only use this call to validate the RPC call, don't add the watch set
-			_, termCfg, err := state.ConfigEntry(nil, structs.TerminatingGateway, args.ServiceName, &args.EnterpriseMeta)
-			if err != nil {
-				return err
+			supportedGateways := []string{structs.IngressGateway, structs.TerminatingGateway}
+			var found bool
+			for _, kind := range supportedGateways {
+				// We only use this call to validate the RPC call, don't add the watch set
+				_, entry, err := state.ConfigEntry(nil, kind, args.ServiceName, &args.EnterpriseMeta)
+				if err != nil {
+					return err
+				}
+				if entry != nil {
+					found = true
+				}
 			}
 
-			if ingressCfg == nil && termCfg == nil {
+			if !found {
 				return fmt.Errorf("service %q is not a terminating-gateway or ingress-gateway", args.ServiceName)
```

Comment:
How about:
```suggestion
				return fmt.Errorf("service %q is not a configured terminating-gateway or ingress-gateway", args.ServiceName)
```

Since the gateway registration might exist in the catalog.